src/modules/SignUp/SignUp.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { withRouter } from 'react-router-dom';
+import * as API from '../../API/APIcalls';
+import { makeStyles, ThemeProvider, useTheme } from '@material-ui/core/styles';
+import image from '../../assets/login-landing-back.jpg';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+
+import {
+  FilledInput,
+  Button,
+  InputAdornment,
+  IconButton,
+  Container,
+} from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    background: `url(${image})`,
+    backgroundSize: 'cover',
+    width: '100%',
+    height: '100vh',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'flex-start',
+  },
+  input: {
+    margin: theme.spacing(1.5),
+    width: '30ch',
+  },
+}));
+
+const SignUp = (props) => {
+  let { history, setUser } = props;
+  const theme = useTheme();
+  const classes = useStyles();
+  const [signUpError, setSignUpError] = useState(false);
+  const [duplicateWarning, setduplicateWarning] = useState(false);
+  const [loginError, setLoginError] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const [values, setValues] = useState({
+    name: '',
+    email: '',
+    password: '',
+  });
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+
+  const handleChange = (prop) => (event) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+
+  useEffect(() => {
+    if (!values.name || !values.email || !values.password) {
+      setDisabled(true);
+      return;
+    }
+    setDisabled(false);
+  }, [values.name, values.email, values.password]);
+
+  const loginUser = () => {
+    API.fetchUserLogin(values.email, values.password)
+      .then((response) => {
+        setUser({
+          id: response.id,
+          name: response.name,
+          email: response.email,
+          token: `Bearer ${response.token}`,
+        });
+      })
+      .then(() => history.push('/dashboard'))
+      .catch((error) => setLoginError(true));
+  };
+
+  const createUser = () => {
+    setSignUpError(false);
+    setduplicateWarning(false);
+    API.createNewUser(values.name, values.email, values.password).then(
+      (response) => {
+        if (response.status === 409) {
+          setDisabled(true);
+          setduplicateWarning(true);
+          return;
+        } else if (response.email === values.email) {
+          loginUser(values.email, values.password);
```

Comment:
@Shawntru here is where I added the login, this seems to fix the problem of the new user not being signed in and not adding the new dreams on first login.